File: common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/ProductionEvent.scala
package uk.ac.wellcome.models.work.internal

case class ProductionEvent[+DataId](
  label: String,
  places: List[Place[DataId]],
  agents: List[AbstractAgent[DataId]],
  dates: List[Period[DataId]],
  function: Option[Concept[DataId]],
  ontologyType: String = "ProductionEvent"
)

File: catalogue_graph/src/ingestor/models/display/production_event.py
from pydantic import BaseModel

from .concept import DisplayConcept


class DisplayProductionEvent(BaseModel):
    label: str
    places: list[DisplayConcept]
    agents: list[DisplayConcept]
    dates: list[DisplayConcept]
    function: DisplayConcept | None
    type: str = "ProductionEvent"
